src/Symfony/Contracts/Deprecation/README.md

```diff
@@ -0,0 +1,22 @@
+Symfony Deprecation Contracts
+=============================
+
+A generic function and convention to trigger deprecation notices.
+
+This package provides a single global function named `deprecated()`.
+Its purpose is to trigger deprecations in a way that can be silenced on production environments
+by using the `zend.assertions` ini setting and that can be caught during development to generate reports.
+
+The function requires at least 3 arguments:
+ - the name of the package that is triggering the deprecation
+ - the version of the package that introduced the deprecation
+ - the message of the deprecation
+ - more arguments can be provided: they will be inserted in the message using `printf()` formatting
```

Comment:
I would remove this possibility as it's not part of the signature.

Using `deprecated('symfony/blockchain', 8.9, sprintf('Using "%s" is deprecated, use "%s" instead.', 'bitcoin', 'fabcoin'));` seems readable enough.

Reply:
It could be part of the signature by using a variadic argument. Would you be ok with that or would you always prefer people use `sprintf`?

Reply:
I think using sprintf() is frequent enough to be worth being builtin.
Signature updated with an explicit variadic argument, does this fix your concern?

Reply:
Having an explicit argument works for me.